### PR TITLE
Refactor trip controller

### PR DESCRIPTION
**Pull request by IgorrMagalhaess**
## Type of change
- [ ] New feature
- [ ] Bug Fix
- [ ] Test
- [x] Refactor/Style

## Description
Relevant User Story(s):
*Brief summary of the change(s) included in this pull request*
- Include logic in controller to handle different user_id
- test for only showing trips for authorized user
- test for only showing trip details for authorized user
- test only updating trip details for authorized user
- helper method in application controller to handle user error
- if statements in actions to check for user_id before allowing rendering index, show, and update.
- refactor to include callback functions and reduce controller logic

This description closes #

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

## Fun fact or favorite .gif:
Did you know that octopuses have three hearts? Two of their hearts pump blood to their gills, while the third pumps it to the rest of their body.

### CODE REVIEW

File: app/controllers/api/v1/trips_controller.rb [hunk 1]
```diff
@@ -25,4 +26,16 @@ def create
...
+      @trip = Trip.find(params[:id])
+   end
+
+   def confirm_user
+      render_user_error(params) if @trip.user_id != params[:user_id].to_i 
+   end
+
+   def filter_user_trips
+      @trips = Trip.where(user_id: params[:user_id])
+   end
```
Comment (yainporter) [inline app/controllers/api/v1/trips_controller.rb:46]: Should this be a model method?

File: app/controllers/api/v1/trips_controller.rb [hunk 2]
```diff
@@ -1,18 +1,19 @@
 class Api::V1::TripsController < ApplicationController
+   before_action :set_trip, only: [:show, :update]
+   before_action :confirm_user, only: [:show, :update]
+   before_action :filter_user_trips, only: [:index]
+
```
Comment (yainporter) [inline app/controllers/api/v1/trips_controller.rb:5]: What does the only: part do?

File: app/controllers/api/v1/trips_controller.rb (hunk 1 above)
Comment (yainporter) [reply app/controllers/api/v1/trips_controller.rb:46]: Since it is using .where

Comment (IgorrMagalhaess) [reply app/controllers/api/v1/trips_controller.rb:46]: I think something simple like this is not necessary. Only if it was a more elaborated query, but I can def put this in the model if you think it's more appropriate.

File: app/controllers/api/v1/trips_controller.rb (hunk 2 above)
Comment (IgorrMagalhaess) [reply app/controllers/api/v1/trips_controller.rb:5]: I'm not sure I understand your question

Comment (hoadam) [reply app/controllers/api/v1/trips_controller.rb:5]: this means the filte_user_trips only work on index method, not all of the methods in the controller

Comment (yainporter) [reply app/controllers/api/v1/trips_controller.rb:5]: Ahhh, I see, thanks

File: app/controllers/api/v1/trips_controller.rb (hunk 1 above)
Comment (IgorrMagalhaess) [reply app/controllers/api/v1/trips_controller.rb:46]: Just created a model method

**Review by yainporter — APPROVED** [review]
Nice job refactoring and making sure sad paths are tested! Something to think about for the future: Can a Trip still be destroyed if it has DailyItinerary attached?

**Review by Selena730 — APPROVED** [review]

**Pull request merged**